tests/pg_session_jwt.rs

```diff
@@ -0,0 +1,214 @@
+    let now = SystemTime::now()
+        .duration_since(UNIX_EPOCH)
+        .unwrap()
+        .as_secs();
+
+    let jwt = sign_jwt(&sk, r#"{"kid":1}"#, json!({"jti": 1, "nbf": now + 10}));
+
+    tx.execute("select auth.init()", &[])?;
+    tx.execute("select auth.jwt_session_init($1)", &[&jwt])?;
+
+    Ok(())
+}
+
+fn invalid_exp(sk: &SigningKey, tx: &mut postgres::Client) -> Result<(), postgres::Error> {
+    let now = SystemTime::now()
+        .duration_since(UNIX_EPOCH)
+        .unwrap()
+        .as_secs();
+
+    let jwt = sign_jwt(
+        &sk,
+        r#"{"kid":1}"#,
+        json!({"jti": 1,  "nbf": now - 10, "exp": now - 5}),
+    );
+
+    tx.execute("select auth.init()", &[])?;
+    tx.execute("select auth.jwt_session_init($1)", &[&jwt])?;
+
+    Ok(())
+}
+
+fn valid_time(sk: &SigningKey, tx: &mut postgres::Client) -> Result<(), postgres::Error> {
+    let now = SystemTime::now()
+        .duration_since(UNIX_EPOCH)
+        .unwrap()
+        .as_secs();
+
+    let header = r#"{"kid":1}"#;
+    let jwt1 = sign_jwt(
+        &sk,
+        header,
+        json!({"jti": 1, "nbf": now - 10, "exp": now + 10}),
+    );
+    let jwt2 = sign_jwt(&sk, header, json!({"jti": 2, "nbf": now - 10}));
+    let jwt3 = sign_jwt(&sk, header, json!({"jti": 3, "exp": now + 10}));
+
+    tx.execute("select auth.init()", &[])?;
+    tx.execute("select auth.jwt_session_init($1)", &[&jwt1])?;
+    tx.execute("select auth.jwt_session_init($1)", &[&jwt2])?;
+    tx.execute("select auth.jwt_session_init($1)", &[&jwt3])?;
+
+    Ok(())
+}
+
+fn test_pg_session_jwt(sk: &SigningKey, tx: &mut postgres::Client) -> Result<(), postgres::Error> {
+    let header = r#"{"kid":1}"#;
+    let jwt1 = sign_jwt(&sk, header, r#"{"sub":"foo","jti":1}"#);
+    let jwt2 = sign_jwt(&sk, header, r#"{"sub":"bar","jti":2}"#);
+
+    tx.execute("select auth.init()", &[])?;
+    tx.execute("select auth.jwt_session_init($1)", &[&jwt1])?;
+    let user_id = tx.query_one("select auth.user_id()", &[])?;
+    let user_id = user_id.get::<_, String>("user_id");
+    assert_eq!(user_id, "foo");
+
+    tx.execute("select auth.jwt_session_init($1)", &[&jwt2])?;
+    let user_id = tx.query_one("select auth.user_id()", &[])?;
+    let user_id = user_id.get::<_, String>("user_id");
+    assert_eq!(user_id, "bar");
+
+    Ok(())
+}
+
+// bgworker process exits after execution, because of that we don't need to test case for more
+// than one JWT
+fn test_bgworker(sk: &SigningKey, tx: &mut postgres::Client) -> Result<(), postgres::Error> {
+    let header = r#"{"kid":1}"#;
+    let jwt = sign_jwt(&sk, header, r#"{"sub":"foo","jti":1}"#);
+
+    tx.execute(&format!("set neon.auth.jwt = '{jwt}'"), &[])?;
+    let user_id = tx.query_one("select auth.user_id()", &[])?;
+    let user_id = user_id.get::<_, String>("user_id");
+    assert_eq!(user_id, "foo");
+
+    Ok(())
+}
+
+// fn discard() -> eyre::Result<()> {
```

Comment:
@conradludgate can I remove this commented function or you'd prefer to have it committed?